### PR TITLE
doc(adr): Resource => ApiResource

### PR DESCRIPTION
**Pull request by soyuka**
Update the ADR document with ApiResource instead of Resource as its a PHP reserved word.

### CODE REVIEW

File: docs/adr/0002-resource-definition.md [hunk 1]
```diff
@@ -24,17 +24,17 @@ In API Platform, this resource identifier is also named [IRI (Internationalized
 ```php
 <?php
 
-#[Resource]
+#[ApiResource]
 class Users
 {
-    #[ApiProperty(iri="hydra:member")]
+    #[ApiProperty(types=["hydra:member"])]
```
Comment (dunglas) [inline docs/adr/0002-resource-definition.md:30]:
```suggestion
    #[ApiProperty(types="hydra:member")]
```

(This works, I already merged the patch, arrays are still supported too)

File: docs/adr/0002-resource-definition.md [hunk 2]
```diff
@@ -24,17 +24,17 @@ In API Platform, this resource identifier is also named [IRI (Internationalized
...
     public User[] $member = [];
```
Comment (dunglas) [inline docs/adr/0002-resource-definition.md:31]:
```suggestion
    public irable $member = [];
```

We should update the PHPDoc accordingly.

File: docs/adr/0003-uri-variables.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  public Company $company;
+}
+```
+
+To make this work, API Platform needs to know what property of the class User has the link to Company. 
+
+## Considered Options
+
+* Keeping the tuple but adding a third value which is the property referenced by the parent class.
+* Using a map to allow granularity configuration of the URI variables
```
Comment (dunglas) [inline docs/adr/0003-uri-variables.md:30]:
```suggestion
* Using a map to allow granularity configuration of the URI variables.
```

File: docs/adr/0003-uri-variables.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    ]
+]
+```
+
+Where `uriTemplate` keys are the URI template's variable names. Its value is a map where:
+
+- `class` is the PHP FQDN to the class this value belongs to
+- `identifiers` are the properties of the class to which we map the URI variable
+- `composite_identifier` is used to match a single variable to multiple identifiers (`ida=1;idb=2` to `class::ida` and `class::idb`)
+- `property` represents the property that has a link to the next identifier
```
Comment (dunglas) [inline docs/adr/0003-uri-variables.md:56]:
```suggestion
- `property` represents the property that has a link to the next identifier

As of PHP 8.1, PHP will support [nested attributes](https://wiki.php.net/rfc/new_in_initializers). We'll introduce a proper class as an alternative to the associative array when PHP 8.1 will be released.
```

**Review by dunglas — CHANGES_REQUESTED** [review]

File: docs/adr/0002-resource-definition.md (hunk 2 above)
Comment (norkunas) [reply docs/adr/0002-resource-definition.md:31]: maybe `iterable` :slightly_smiling_face:

Comment (dunglas) [reply docs/adr/0002-resource-definition.md:31]: Fixed

**Pull request merged**